variety/AddWallhavenDialog.py

```diff
@@ -0,0 +1,192 @@
+# -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
+### BEGIN LICENSE
+# Copyright (c) 2012, Peter Levi <peterlevi@peterlevi.com>
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranties of
+# MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+### END LICENSE
+import urllib
+from urllib.parse import urlencode
+
+from gi.repository import Gdk, Gtk  # pylint: disable=E0611
+
+from variety.AddConfigurableDialog import AddConfigurableDialog
+from variety.plugins.builtin.downloaders.WallhavenDownloader import WallhavenDownloader
+from variety.Util import Util
+from variety_lib.helpers import get_builder
+
+
+class AddWallhavenDialog(AddConfigurableDialog):
+    __gtype_name__ = "AddWallhavenDialog"
+
+    def __new__(cls):
+        builder = get_builder("AddWallhavenDialog")
+        new_object = builder.get_object("add_wallhaven_dialog")
+        new_object.finish_initializing(builder)
+        return new_object
+
+    def finish_initializing(self, builder):
+        """Called when we're finished initializing.
+
+        finish_initalizing should be called after parsing the ui definition
+        and creating a WelcomeDialog object with it in order to
+        finish initializing the start of the new WelcomeDialog
+        instance.
+        """
+        # Get a reference to the builder and set up the signals.
+        self.builder = builder
+        self.ui = builder.get_ui(self)
+        self.edited_row = None
+        self.load_ui()
+
+    def set_source(self, source):
```

Comment:
I'm not sure it's worth complicating things by defining get_ui_use_apikey_or_not_text, get_api_key_instruction, get_ui_apikey_instruction in the source and then having to override set_source here just to pull them from there. They are "static" texts so we can just hardcode them in the AddWallhavenDialog UI we are adding. The reason for the instruction/short_instruction being defined in the source is because we use a common "generic" dialog for most  configurable sources (incl. for wallhaven up till now), so we can't hardcode them in the UI. But here we are just complicating the code for these.